docs/apache-airflow-providers/howto/create-update-providers.rst

```diff
@@ -0,0 +1,322 @@
+
+      NewProviderOperator
+      ===================
+
+      Use the :class:`~airflow.providers.new_provider.operators.NewProviderOperator` to do something
+      amazing with Airflow!
+
+      Using the Operator
+      ^^^^^^^^^^^^^^^^^^
+
+      The NewProviderOperator requires a ``connection_id`` and this other awesome parameter.
+      You can see an example below:
+
+      .. exampleinclude:: /../../airflow/providers/new_provider/example_dags/example_new_provider.py
+          :language: python
+          :start-after: [START howto_operator_new_provider]
+          :end-before: [END howto_operator_new_provider]
+
+
+In the ``docs/apache-airflow-providers-new_provider/commits.rst``:
+
+- add the initial information of your providers. This file is updated automatically by Airflow. Below is shown an example.
+
+  .. code-block:: RST
+
+      package apache-airflow-providers-new_provider
+      ---------------------------------------------
+
+      `New_Provider <https://example.io/>`__
+
+
+      This is detailed commit list of changes for versions provider package: ``new_provider``.
+      For high-level changelog, see :doc:`package information including changelog <index>`.
+
+In the ``docs/apache-airflow-providers-new_provider/index.rst``:
+
+- add all information of the purpose of your provider. It is recommended to check with another provider to help you complete this document as best as possible.
+
+
+In the ``/airflow/providers/new_provider/CHANGELOG`` add the initial information. Providers start with 1.0.0 version.
+
+  .. code-block:: RST
+
+      Changelog
+      ---------
+
+      1.0.0
+      .....
+
+      Initial version of the provider.
+
+In the ``airflow/providers/new_provider/provider.yaml`` add information of your provider:
+
+  .. code-block:: yaml
+
+      package-name: apache-airflow-providers-new_provider
+      name: New_Provider
+      description: |
+        `New_Provider <https://example.io/>`__
+      versions:
+        - 1.0.0
+
+      integrations:
+        - integration-name: New_Provider
+          external-doc-url: https://www.example.io/
+          logo: /integration-logos/new_provider/New_Provider.png
+          how-to-guide:
+            - /docs/apache-airflow-providers-new_provider/operators/new_provider.rst
+          tags: [service]
+
+      operators:
+        - integration-name: New_Provider
+          python-modules:
+            - airflow.providers.new_provider.operators.new_provider
+
+      hooks:
+        - integration-name: New_Provider
+          python-modules:
+            - airflow.providers.new_provider.hooks.new_provider
+
+      sensors:
+        - integration-name: New_Provider
+          python-modules:
+            - airflow.providers.new_provider.sensors.new_provider
+
+      hook-class-names:
+        - airflow.providers.new_provider.hooks.new_provider.NewProviderHook
+
+After changing and creating these files you can build the documentation locally. The two commands below will
```

Comment:
```suggestion
You only need to add `hook-class-names` in case you have some hooks that have customized UI behavior. 

After changing and creating these files you can build the documentation locally. The two commands below will
```

Also reference to http://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#custom-connection-types would be nice.